src/components/ListItem.jsx

```diff
@@ -1,5 +1,58 @@
 import './ListItem.css';
+import { useState, useEffect } from 'react';
+import { updateItem } from '../api/firebase';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({
+	name,
+	itemId,
+	dateLastPurchased,
+	totalPurchases,
+	listId,
+}) {
+	const [isChecked, setIsChecked] = useState(false);
+
+	const increasePurchases = totalPurchases + 1;
+	const decreasePurchases = totalPurchases - 1;
+
+	useEffect(() => {
```

Comment:
I refactored the useEffect slightly for better readability: 
```
useEffect(() => {
		const checkLastPurchaseStatus = () => {
			const currentTime = Date.now();

			if (dateLastPurchased === null) {
				return;
			}

			const timeSinceLastPurchase = currentTime - dateLastPurchased.toMillis();
			setIsChecked(timeSinceLastPurchase < 24 * 60 * 60 * 1000);
		};

		checkLastPurchaseStatus();
	}, [dateLastPurchased]);
```

Reply:
Thank you! We've implemented this suggestion.